File: sdkwork-ai-sdk-typescript/src/api/message.ts
import { aiApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { CountClaudeTokensPostRequest, CountClaudeTokensPostResponse, CreateClaudeMessageRequest, CreateClaudeMessageResponse } from '../types';


export class MessageApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** Count Claude tokens */
  async countClaudeTokens(body: CountClaudeTokensPostRequest): Promise<CountClaudeTokensPostResponse> {
    return this.client.post<CountClaudeTokensPostResponse>(aiApiPath(`/messages/count_tokens`), body);
  }

/** Create Claude message */
  async createClaude(body: CreateClaudeMessageRequest, headers?: Record<string, string>): Promise<CreateClaudeMessageResponse> {
    return this.client.post<CreateClaudeMessageResponse>(aiApiPath(`/messages`), body, undefined, headers);
  }
}

export function createMessageApi(client: HttpClient): MessageApi {
  return new MessageApi(client);
}

File: sdkwork-ai-sdk-go/api/message.go
package api

import (
    sdktypes "github.com/sdkwork/ai-sdk/types"
    sdkhttp "github.com/sdkwork/ai-sdk/http"
)

type MessageApi struct {
    client *sdkhttp.Client
}

func NewMessageApi(client *sdkhttp.Client) *MessageApi {
    return &MessageApi{client: client}
}

// Count Claude tokens
func (a *MessageApi) CountClaudeTokens(body sdktypes.CountClaudeTokensPostRequest) (sdktypes.CountClaudeTokensPostResponse, error) {
    raw, err := a.client.Post(AiApiPath("/messages/count_tokens"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.CountClaudeTokensPostResponse
        return zero, err
    }
    return decodeResult[sdktypes.CountClaudeTokensPostResponse](raw)
}

// Create Claude message
func (a *MessageApi) CreateClaude(body sdktypes.CreateClaudeMessageRequest, headers map[string]string) (sdktypes.CreateClaudeMessageResponse, error) {
    raw, err := a.client.Post(AiApiPath("/messages"), body, nil, headers, "")
    if err != nil {
        var zero sdktypes.CreateClaudeMessageResponse
        return zero, err
    }
    return decodeResult[sdktypes.CreateClaudeMessageResponse](raw)
}
